File: java-client/src/main/java/com/bolour/boardgame/client/message/StartGameResponse.java
/*
 * Copyright 2017-2018 Azad Bolour
 * Licensed under GNU Affero General Public License v3.0 -
 *   https://github.com/azadbolour/boardgame/blob/master/LICENSE.md
 */

package com.bolour.boardgame.client.message;

import com.bolour.boardgame.client.domain.GameParams;
import com.bolour.boardgame.client.domain.PiecePoint;
import com.bolour.boardgame.client.domain.Piece;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public class StartGameResponse {

    public final String gameId;
    public final GameParams gameParams;
    public final List<PiecePoint> boardPiecePoints;
    public final List<Piece> trayPieces;

    @JsonCreator
    public StartGameResponse(
      @JsonProperty("gameId") String gameId,
      @JsonProperty("gameParams") GameParams gameParams,
      @JsonProperty("boardPiecePoints") List<PiecePoint> boardPiecePoints,
      @JsonProperty("trayPieces") List<Piece> trayPieces
    ) {
        this.gameId = gameId;
        this.gameParams = gameParams;
        this.boardPiecePoints = boardPiecePoints;
        this.trayPieces = trayPieces;
    }
    
}

File: scala-server/app/com/bolour/boardgame/scala/common/message/StartGameResponse.scala
/*
 * Copyright 2017-2018 Azad Bolour
 * Licensed under GNU Affero General Public License v3.0 -
 *    https://github.com/azadbolour/boardgame/blob/master/LICENSE.md
 */
package com.bolour.boardgame.scala.common.message

import com.bolour.boardgame.scala.common.domain.{GameParams, PiecePoint, Piece}

/**
  * API response for starting a game.
  *
  * @param gameId Unique id of the game.
  * @param gameParams The game parameters. In case some are assigned at the server.
  * @param boardPiecePoints The initial state of the board. TODO. Unnecessary.
  * @param trayPieces The user's initial tray  pieces.
  */
case class StartGameResponse(
  gameId: String,
  gameParams: GameParams,
  boardPiecePoints: List[PiecePoint],
  trayPieces: List[Piece]
)
